Example/Tests/GINIMultipageReviewControllerTests.swift

```diff
@@ -0,0 +1,187 @@
+                       "Second cell image should match the one passed in the initializer")
+        XCTAssertEqual(secondCell?.pageIndicator.text, "2",
+                       "Second cell indicator should match its position")
+        XCTAssertEqual(thirdCell?.documentImage.image, imageDocuments[2].previewImage,
+                       "Third cell image should match the one passed in the initializer")
+        XCTAssertEqual(thirdCell?.pageIndicator.text, "3",
+                       "Third cell indicator should match its position")
+        XCTAssertEqual(firstCell?.documentImage.contentMode, UIViewContentMode.scaleAspectFill,
+                       "First cell content mode should match the one passed in the initializer")
+    }
+    
+    func testBottomCollectionCellSize() {
+        let vc = MultipageReviewController(imageDocuments: imageDocuments)
+        _ = vc.view
+        vc.view.setNeedsLayout()
+        vc.view.layoutIfNeeded()
+        
+        let firstCellIndexPath = IndexPath(row: 0, section: 0)
+        let cellSize = vc.collectionView(vc.bottomCollection,
+                                         layout: vc.bottomCollection.collectionViewLayout,
+                                         sizeForItemAt: firstCellIndexPath)
+        
+        let size = MultipageReviewBottomCollectionCell.size
+        
+        XCTAssertEqual(cellSize, size,
+                       "Bottom collection cells should have the value declared in the class")
+    }
+    
+    func testBottomCollectionInsets() {
+        let collectionInsets = vc.collectionView(vc.bottomCollection,
+                                                 layout: vc.bottomCollection.collectionViewLayout,
+                                                 insetForSectionAt: 0)
+        
+        XCTAssertEqual(collectionInsets, vc.bottomCollectionInsets,
+                       "Main collection insets should be zero")
+    }
+    
+    func testToolBarItemsOnInitialization() {
+        XCTAssertEqual(self.vc.toolBar.items![0], self.vc.rotateButton,
+                       "First toolbar item should be the rotateButton")
+        XCTAssertEqual(self.vc.toolBar.items![2], self.vc.reorderButton,
+                       "Third toolbar item should be the rotateButton")
+        XCTAssertEqual(self.vc.toolBar.items![4], self.vc.deleteButton,
+                       "Fifth toolbar item should be the rotateButton")
+    }
+    
+    func testToolBarAndBottomCollectionContainerColors() {
+        XCTAssertEqual(vc.toolBar.barTintColor, vc.bottomCollectionContainer.backgroundColor,
+                       "toolbar and bottom collection container background colors should match")
+    }
+    
+    func testDatasourceOnDelete() {
+        let vc = MultipageReviewController(imageDocuments: imageDocuments)
+        _ = vc.view
+        vc.view.setNeedsLayout()
+        vc.view.layoutIfNeeded()
+        
+        (vc.deleteButton.customView as? UIButton)?.sendActions(for: .touchUpInside)
```

Comment:
Why did you choose to send a touch event and not just invoke the Action method? Just curious...

Reply:
To check that the method is trigger after tapping the button. It is a double check.

Reply:
If that's what you prefer, I'm fine with it.